pallets/asset/src/lib.rs

```diff
@@ -1720,15 +1721,14 @@ impl<T: Trait> Module<T> {
         <BalanceOfAtScope<T>>::insert(scope_id, did, updated_balance);
     }
 
-    pub fn ensure_pia_or_owner(
-        ticker: &Ticker,
-        did: IdentityId,
-    ) -> Result<IdentityId, DispatchError> {
-        Self::token_details(&ticker)
-            .primary_issuance_agent
-            .filter(|pia| *pia == did)
-            .or_else(|| Self::is_owner(&ticker, did).then(|| did))
-            .ok_or_else(|| Error::<T>::Unauthorized.into())
+    /// Ensure that `did` is either the token owner, or the assigned PIA
+    /// Returns the PIA
+    pub fn ensure_pia(ticker: &Ticker, did: IdentityId) -> Result<IdentityId, DispatchError> {
+        let pia = Self::token_details(&ticker).primary_issuance_agent;
+        if did == pia {
+            return Ok(pia);
+        }
+        return Err(Error::<T>::Unauthorized.into());
```

Comment:
```suggestion
    pub fn ensure_pia(ticker: &Ticker, did: IdentityId) -> Result<IdentityId, DispatchError> {
        let pia = Self::token_details(&ticker).primary_issuance_agent;
        ensure!(did == pia, Error::<T>::Unauthorized);
        Ok(())
```
No need to return `did` back; zero new information is gained by that.